### ISSUE TITLE
[UrT4.2] userinfo string overlow

### ISSUE THREAD

**Issue by danielepantaleone** [BigBrotherBot/big-brother-bot#346]
I found this while inspecting my B3 log file:

```
150717 16:26:40 DEBUG    'Urt42 Client Connected: [1] aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa - C3D67CD253EC422662476D713A258EE2 ({})'
```

which resulted in:

```
150717 16:26:40 DEBUG    "Storage: setClientAlias data {'num_used': 1, 'alias': 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ', 'time_add': 1437143200, 'time_edit': 1437143200, 'client_id': 7443}"
150717 16:26:40 ERROR    'STDERR "/home/servers/resources/bigbrotherbot/b3_run/pymysql/cursors.py:134: Warning: Data truncated for column \'alias\' at row 1\\n"'
```

This error is also propagated to the `clients` table, hence the client entry is not saved in the database (which is very bad).

The bug is actually not in B3 but in the UrT 4.2 gamecode, since the maximum client name length should be fixed at *32* characters, but I guess (by reading those log lines) it is possible to exploit this limitation by manually setting the client name in the `q3config.cfg` file. Since UrT4.2 won't be updated anymore (as far as I know) I guess we need to address this issue in B3 somehow. I can think of 3 solutions:

1. completely deny client connection when the `name` field is bigger than 32 characters (this will also prevent B3 from failing in storing the client alias)
2. truncate the client name to 32 characters (may generate inconsistence)
3. change the size of the name field in B3 database (there is no fixed value so I guess it won't solve the problem)

I guess solution number 1 is the one which would fix the problem. Opinions?

**Comment by thomasleveil** [comment]
you mean kick the client?

**Comment by danielepantaleone** [comment]
Yes

Inviato da iPhone

**Comment by thomasleveil** [comment]
I would go for:

- by default: truncate name to 32
- if explicitly activated in the main config file: kick + clear log message explaining the kick + explanation message ingame

**Comment by danielepantaleone** [comment]
Fair enough :smile:

**Comment by BidyBiddle** [comment]
We'll try and fix this for UrT 4.3, thanks :+1:
